tests/system_suite_test.go

```diff
@@ -0,0 +1,95 @@
+//go:build !unit
+// +build !unit
+
+package tests
+
+import (
+	"flag"
+	"path"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	apiext "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	k8sRuntime "k8s.io/apimachinery/pkg/runtime"
+	ctlr "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+)
+
+func TestNGF(t *testing.T) {
+	flag.Parse()
+	if *gatewayAPIVersion == "" {
+		panic("Gateway API version must be set")
+	}
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "NGF System Tests")
```

Comment:
how will individual tests be able to change parameters of NGF deployment? i
Example:
- image
- number of replicas

Same with the Gateway API version.
Imagine an upgrade test.  
gw api version X, ngf version A -> gw version Y, ngf version B

Reply:
That would likely require separate suites. The structure of this will become more clear as we start writing and organizing the tests.